query-node/mappings/src/bounty.ts

```diff
@@ -187,6 +194,38 @@ function endWorkingPeriod(store: DatabaseManager, bounty: Bounty, blockNumber: n
   return store.save<Bounty>(bounty)
 }
 
+type JudgmentEntries = [EntryId, OracleWorkEntryJudgment][]
+async function endJudgmentPeriod(
+  store: DatabaseManager,
+  bounty: Bounty,
+  entryJudgments: JudgmentEntries
+): Promise<void> {
+  // Update the bounty status
+  const hasWinners = entryJudgments.some(([, judgment]) => judgment.isWinner)
+  bounty.updatedAt = new Date()
+  bounty.stage = BountyStage[hasWinners ? 'Successful' : 'Failed']
+  await store.save<Bounty>(bounty)
+
+  // Update winner entries status
+  await Promise.all(
+    bounty.entries?.map((entry) => {
+      const judgment = entryJudgments.find(([entryId]) => String(entryId) === entry.id)?.[1]
+
+      if (!judgment) {
+        entry.status = new BountyEntryStatusPassed()
```

Comment:
You shouldn't change the status of the entries that are already in `BountyEntryStatusWithdrawn`, otherwise in Pioneer 2 there is a `Withdraw Stake` button for those entries which doesn't work:
![image](https://user-images.githubusercontent.com/19978260/160093341-98405fca-18e3-4399-83f8-5621bb38b465.png)